docs/docs/tutorial/creating-a-source-plugin/part-2/index.mdx

```diff
@@ -0,0 +1,666 @@
+
+    </Announcement>
+
+1. Scroll up in `source-nodes.ts` to the place where you're calling `console.dir()` and remove that log. Replace it with a destructuring assignment of `data` to get the `posts` and `authors` results out of it (you can also set an empty array as a default value):
+
+    ```ts:title=plugin/src/source-nodes.ts
+    // Rest of the file
+
+    export const sourceNodes: GatsbyNode[`sourceNodes`] = async (gatsbyApi) => {
+      // Rest of sourceNodes
+
+      const { posts = [], authors = [] } = data
+    }
+    ```
+
+    So both `posts` and `authors` are an array of objects that you should iterate over. Each object will become a Gatsby node. Iterate over both arrays and call `nodeBuilder` inside the loops:
+
+    ```ts:title=plugin/src/source-nodes.ts
+    // Rest of the file
+
+    export const sourceNodes: GatsbyNode[`sourceNodes`] = async (gatsbyApi) => {
+      // Rest of sourceNodes
+
+      const { posts = [], authors = [] } = data
+
+      // highlight-start
+      for (const post of posts) {
+        nodeBuilder({ gatsbyApi, input: { type: NODE_TYPES.Post, data: post } })
+      }
+
+      for (const author of authors) {
+        nodeBuilder({ gatsbyApi, input: { type: NODE_TYPES.Author, data: author } })
+      }
+      // highlight-end
+    }
+    ```
+
+You can see the benefit of creating a helper function: You don't have to repeat yourself inside the loops, making the whole code more concise and clear.
+
+## Display results in your site
+
+You've created GraphQL nodes from your data, time to display them also in your site.
+
+Restart the `develop:site` script in your terminal. Once the development server is running, go to Gatsby's GraphiQL IDE (Need a refresher? Read the [introduction to GraphiQL](/docs/tutorial/part-4/#use-graphiql-to-explore-the-data-layer-and-write-graphql-queries)) at `http://localhost:8000/___graphql`.
+
+Inside GraphiQL, try a query against your newly created `Author` and `Post` nodes, for example:
+
+```graphql
+query MyQuery {
+  allAuthor {
+    nodes {
+      name
+    }
+  }
+  allPost {
+    nodes {
+      title
+      slug
+      image {
+        url
+      }
+      author
+    }
+  }
+}
+```
+
+You should see a result in your browser:
+
+![A screenshot of "localhost:8000/___graphql" in a web browser. The GraphiQL IDE shows the results of the GraphQL query mentioned above.](./02-graphiql-interface.png)
+
+Great, you successfully queried your data in a Gatsby site! As the last step of this part, create pages from all available posts and also display them on the index page.
+
+First, edit the index page to display all posts and link to their individual pages (which you'll create in just a second):
+
+```tsx:title=site/src/pages/index.tsx
+import * as React from "react"
+import { Link, graphql, HeadFC, PageProps } from "gatsby"
+
+export default function IndexPage({
+  data: {
+    allPost: { nodes: posts },
+  },
+}: PageProps<Queries.IndexPageQuery>): React.ReactElement {
```

Comment:
`Queries.IndexPageQuery` is flagged by TS as red for me when pasting this in at first.

Giving VSCode a minute to catch up worked... or maybe it was the trick of removing the `y` in `IndexPageQuery` and letting VSCode find the import. Don't know if this warrants any call out however.